examples/gno.land/r/nt/boards2/v1/freeze.gno

```diff
@@ -0,0 +1,76 @@
+package boards2
+
+import "std"
+
+type freezeArgs struct {
+	boardID  BoardID
+	threadID PostID
+	replyID  PostID
+}
+
+func (args freezeArgs) getThread() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	return thread, board
+}
+
+func (args freezeArgs) getReply() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	assertThreadVisible(thread)
+	assertThreadIsNotFrozen(thread)
+
+	reply := mustGetReply(thread, args.replyID)
+	assertReplyVisible(reply)
+	return reply, board
+}
+
+func setReplyReadOnly(args freezeArgs, isReadOnly bool) {
+	reply, board := args.getReply()
+
+	if isReadOnly {
+		// disallow freezing already frozen reply
+		assertReplyIsNotFrozen(reply)
+	}
+
+	caller := std.OriginCaller()
+	permArgs := Args{args.boardID, args.threadID, args.replyID}
+	board.perms.WithPermission(caller, PermissionReplyFreeze, permArgs, func(Args) {
+		reply, _ := args.getReply()
+		reply.SetReadOnly(isReadOnly)
+	})
+}
+
+func setThreadReadOnly(args freezeArgs, isReadOnly bool) {
+	thread, board := args.getThread()
+
+	if isReadOnly {
+		// disallow freezing of already frozen thread
+		assertThreadIsNotFrozen(thread)
+	}
+
+	caller := std.OriginCaller()
+	permArgs := Args{args.boardID, args.threadID}
+	board.perms.WithPermission(caller, PermissionThreadFreeze, permArgs, func(Args) {
+		thread, _ := args.getThread()
+		thread.SetReadOnly(isReadOnly)
+	})
+}
+
+func setBoardReadOnly(boardID BoardID, isReadOnly bool) {
+	board := mustGetBoard(boardID)
+	if isReadOnly {
+		assertBoardIsNotFrozen(board)
+	}
```

Comment:
This could potentially be removed from here and left in `FreezeBoard()` as it's only going to be used there, it would remove the condition making this function smaller, so it would look like:
```go
func FreezeBoard(boardID BoardID) {
  board := mustGetBoard(boardID)
  assertBoardIsNotFrozen(board)

  setBoardReadOnly(boardID, true)
}
```